src/System.Net.Http/src/netcore50/System/Net/HttpClientHandler.cs

```diff
@@ -522,6 +469,82 @@ private void ApplyDecompressionSetting(HttpRequestMessage request, Decompression
                 _rtFilter.ClientCertificate = clientCertificates[0];
             }
         }
+
+        private RTPasswordCredential RTPasswordCredentialFromICredentials(ICredentials creds)
+        {
+            // The WinRT PasswordCredential object does not have a special credentials value for "default credentials".
+            // In general, the UWP HTTP platform automatically manages sending default credentials, if no explicit
+            // credential was specificed, based on if the app has EnterpriseAuthentication capability and if the endpoint
+            // is listed in an intranet zone.
+            //
+            // A WinRT PasswordCredential object that is either null or created with the default constructor (i.e. with
+            // empty values for username and password) indicates that there is no explicit credential. And that means
+            // that the default logged-on credentials might be sent to the endpoint.
+            //
+            // There is currently no WinRT API to turn off sending default credentials other than the capability
+            // and intranet zone checks described above. In general, the UWP HTTP model for specifying default
+            // credentials is orthogonal to how the .NET System.Net APIs have been designed.
+            if (creds == null || creds == CredentialCache.DefaultCredentials)
+            {
+                return null;
+            }
+            else
+            {
+                Debug.Assert(creds is NetworkCredential);
+
+                NetworkCredential networkCred = (NetworkCredential)creds;
+
+                // Creating a new WinRT PasswordCredential object with the default constructor ends up
+                // with empty strings for username and password inside the object. However, one can't assign
+                // empty strings to those properties; otherwise, it will throw an error.
+                RTPasswordCredential rtCreds = new RTPasswordCredential();
+                if (!string.IsNullOrEmpty(networkCred.UserName))
```

Comment:
Instead of calling multiple COM pinvokes why not just decide first if the user/pwd are empty and then call the right RTPasswordCredential ctor?